Server/src/services/dataProcess/JsonFileHandler.ts

```diff
@@ -0,0 +1,200 @@
+import { AbstractFileHandler, FileHandlerFactory } from './IDataProcessService';
+import { DataUploadModel } from '../../models/DataUploadModel'
+import { Authors } from '../../models/entities/Authors';
+import { IMaterials } from '../../models/interfaces/MaterialsInterface';
+import { IAuthors } from '../../models/interfaces/AuthorsInterface';
+import { BadRequest } from '@tsed/exceptions';
+import { validationSchema } from '../helpers/validationSchema';
+import { IJsonDatasetModel } from '../../genericInterfaces/DataProcessInterfaces'
+const fileSystem = require('fs');
+
+
+export class JsonFileFactory extends FileHandlerFactory {
+
+    getFileHandler(filePath: string): AbstractFileHandler {
+        return new JsonFileHandler(filePath)
+    }
+}
+
+export class JsonFileHandler extends AbstractFileHandler {
+
+    async parseFile(): Promise<IJsonDatasetModel> {
+        try {
+            this.parsedFileData = await JSON.parse(fileSystem.readFileSync(this.filePath))
```

Comment:
parsedFileData doesn't seem to be declared at the start of the class. Oversight?